### DISCUSSION TITLE
Add presets on Android without rooting

### DISCUSSION THREAD

**Discussion by thepicklecatxo116** [projectM-visualizer/projectm#574] [Q&A]
Hi I downloaded ProjectM through the play store and upgraded to the 4 dollar version. I was curious and checked this website and saw there was a download link for 40k presets(which I've already downloaded) and I was wondering if there's anyway to add them. I know I could if I were to root my phone but I have a weird version of android that has alot more steps then just pressing a few buttons and I don't want to risk briking my phone for it. I've checked as deep in my files as I can and I can't gain access to the presets folder. Would you happen to know if there's a work-around or a way to gain access to the folder?

**Comment by OfficialIncubo** [comment]
Yeah. You can see your presets that you have on storage/emulated/0/Android/data/com.psperl.projectM/files/presets_dir

**Comment by JunkCreek** [comment]
@OfficialIncubo 
In Android 11+, we can't access Android-Data and Android-Obb Folder, we can only had Access to Android-Media Folder.

Is there any workaround, suc-as make application data using Android-Media or Any Folder instead restricted only in Android-Data Folder?

**Reply by OfficialIncubo** [reply]
Try using [FV File Explorer](https://folderv.com/). This will grant you to access the data folder and OBS folder with the built-in one or with [Shizuku](https://shizuku.rikka.app/).

Will it work for you?

**Reply by JunkCreek** [reply]
Trying your suggestions and it works!

![Screenshot_2025-01-11-14-02-16-18_3177cc2cf37291944719f629e393766d](https://github.com/user-attachments/assets/b8c7ed29-c025-4f19-9f20-01d7f0df074e)

Thankyou very much.

By the way, can dev team make preset dir to be on any folder or any specific 'special folder', eg. Downloads-preset_dir instead in Android-Data folder?

**Reply by kblaschke** [reply]
> By the way, can dev team make preset dir to be on any folder or any specific 'special folder', eg. Downloads-preset_dir instead in Android-Data folder?

Since we're not developing the Android app, you should direct this request to the actual app author. You can find his email in the Play Store.

**Reply by JunkCreek** [reply]
@kblaschke 
Oh, I'm so sorry, I didn't read it on the front page.

Thankyou very much for quick response.

**Comment by JunkCreek** [comment]
@OfficialIncubo 
I've tried using FV File Manager and it works well.

But, since I'm using it, my phone had a kind of 'bugs'.

My File Manager + can't 'detect' any photos/images and videos. So did my PowerAmp, it can't save AlbumArt and even gives missing already downloaded AlbumArts.

So I uninstalled FV and I'm using another File Manager that supports iAdb 'internally', ES Explorer, but unfortunately it isn't on google playstore, so I'm using it with my own risk.

There is another 'work around', using EX File Manager, Install iAdb apk from Play Store. It was almost flawless, no 'bugs' such as happened with FV, but I should 'reconnect' and start iAdb service everytime I restarted my phone, and it was almost a tiresome.

So I'm using ES Explorer when I need access to android-data, and when I'm done, I uninstalled ES just in case in order for having safety 'back'.

![Screenshot_2025-01-12-22-11-07-91_846b44643ec609f507828878741e1f9a](https://github.com/user-attachments/assets/5ad791b6-0704-4fca-8785-4f93d355c924)

**Comment by JesusGod-Pope666-Info** [comment]
I HATE GOOGLE, DESTROYING ANDROID IN ANY VVAY POSSIBLE.

**Comment by JesusGod-Pope666-Info** [comment]
I bought the Android as vvell, but I must say it has a lot of bugs, and it formerly had stars to add to milk's that I vvas using to sort things out for the first 50 demo milks. The paid version sure needs a lot of more love from the programmer. Crashing, forgetting settings and such. Sadly... No idea hovv to make playlist, but i guess it might have something to do vvith the folder in the Android data, but not sure.... You can add files but it still have all files shovvn. And even if I just deleted all 300 milks for adding my ovvn, all of them just came back vvhile toying around vvith the playlist.... just great..... clicking game of madness and nonsense.

but if i can fix the playlist thing, then I could choose my ovvn folder and hopefully have it run from that.
But clearly this app need a lot more love on Android, and the remake or update of it, seemed to not be really finished and done to further polish things, seems more like a beta version or maybe even an alpha version.

- Darkijah

**Comment by JesusGod-Pope666-Info** [comment]
Total Commander novv been totally locked out of data, so sickening evil of FILTHY EVIL GOOGLE. Someone take over from Google so vvee can leave this sinking ship of sicknees - tired of fighting the bullshit, it has been vvay to many years and years and more years. Just trying to do normal things vvith MY DEVICES. Yet Google treats it as it is all theirs.

**Comment by JesusGod-Pope666-Info** [comment]
> Hi I downloaded ProjectM through the play store and upgraded to the 4 dollar version. I was curious and checked this website and saw there was a download link for 40k presets(which I've already downloaded) and I was wondering if there's anyway to add them. I know I could if I were to root my phone but I have a weird version of android that has alot more steps then just pressing a few buttons and I don't want to risk briking my phone for it. I've checked as deep in my files as I can and I can't gain access to the presets folder. Would you happen to know if there's a work-around or a way to gain access to the folder?

just curious, vvhat link are you talking about in regards of 40.000 milk's
